AssemblerLib/MeshComponentMap.cpp

```diff
@@ -48,6 +48,39 @@ MeshComponentMap::MeshComponentMap(
         renumberByLocation();
 }
 
+MeshComponentMap
+MeshComponentMap::getSubset(std::vector<MeshLib::MeshSubsets*> const& components) const
+{
+    // New dictionary for the subset.
+    ComponentGlobalIndexDict subset_dict;
+
+    std::size_t comp_id = 0;
+    for (auto c : components)
+    {
+        if (c == nullptr)   // Empty component
+        {
+            comp_id++;
+            continue;
+        }
+        for (std::size_t mesh_subset_index = 0; mesh_subset_index < c->size(); mesh_subset_index++)
+        {
+            MeshLib::MeshSubset const& mesh_subset = c->getMeshSubset(mesh_subset_index);
+            std::size_t const mesh_id = mesh_subset.getMeshID();
+            // Lookup the locations in the current mesh component map and
+            // insert the full lines into the subset dictionary.
+            for (std::size_t j=0; j<mesh_subset.getNNodes(); j++)
+                subset_dict.insert(getLine(Location(mesh_id,
+                    MeshLib::MeshItemType::Node, mesh_subset.getNodeID(j)), comp_id));
+            for (std::size_t j=0; j<mesh_subset.getNElements(); j++)
+                subset_dict.insert(getLine(Location(mesh_id,
+                    MeshLib::MeshItemType::Cell, mesh_subset.getElementID(j)), comp_id));
+        }
+        comp_id++;
+    }
+
+    return MeshComponentMap(subset_dict);
```

Comment:
At this point, as Tom added, there is no copy because of the RVO, which will be done in the debugging mode too: On lower level of compilation the address of the object from the caller side is passed (like hidden parameter), and then used for object construction here.

But while looking into this potential copy here, I could not avoid the call to the copy ctor. And that call is not here, but in initialization list of `LocalToGlobalIndexMap` [line 43 in the cpp file](https://github.com/ufz/ogs/pull/599/files#diff-3d365cd12eafd459267739cd14224f74R43).
I shall try to get rid of it.